x-pack/plugins/observability/server/lib/rules/slo_burn_rate/executor.ts

```diff
@@ -28,11 +29,78 @@ import {
   BurnRateAllowedActionGroups,
   BurnRateRuleParams,
   BurnRateRuleTypeState,
+  WindowSchema,
 } from './types';
+import { ALERT_ACTION_ID } from '../../../../common/constants';
 
 const SHORT_WINDOW = 'SHORT_WINDOW';
 const LONG_WINDOW = 'LONG_WINDOW';
 
+async function evaluateWindow(slo: SLO, summaryClient: DefaultSLIClient, windowDef: WindowSchema) {
+  const longWindowDuration = new Duration(
+    windowDef.longWindow.value,
+    toDurationUnit(windowDef.longWindow.unit)
+  );
+  const shortWindowDuration = new Duration(
+    windowDef.shortWindow.value,
+    toDurationUnit(windowDef.shortWindow.unit)
+  );
+
+  const sliData = await summaryClient.fetchSLIDataFrom(slo, [
+    { name: LONG_WINDOW, duration: longWindowDuration.add(slo.settings.syncDelay) },
+    { name: SHORT_WINDOW, duration: shortWindowDuration.add(slo.settings.syncDelay) },
+  ]);
+
+  const longWindowBurnRate = computeBurnRate(slo, sliData[LONG_WINDOW]);
+  const shortWindowBurnRate = computeBurnRate(slo, sliData[SHORT_WINDOW]);
+
+  const shouldAlert =
+    longWindowBurnRate >= windowDef.burnRateThreshold &&
+    shortWindowBurnRate >= windowDef.burnRateThreshold;
+
+  return {
+    shouldAlert,
+    longWindowBurnRate,
+    shortWindowBurnRate,
+    longWindowDuration,
+    shortWindowDuration,
+    window: windowDef,
+  };
+}
+
+async function evaluate(slo: SLO, summaryClient: DefaultSLIClient, params: BurnRateRuleParams) {
+  if (params.windows && params.windows.length > 0) {
+    const evalWindow = memoize(async (windowDef: WindowSchema) =>
+      evaluateWindow(slo, summaryClient, windowDef)
+    );
+    for (const windowDef of params.windows) {
+      const result = await evalWindow(windowDef);
+      if (result.shouldAlert) {
+        return result;
+      }
+    }
+    // If none of the previous windows match, we need to return the last window
+    // for the recovery context. Since evalWindow is memoized, it shouldn't make
+    // and additional call to evaulateWindow.
+    return await evalWindow(last(sortedWindows) as WindowSchema);
```

Comment:
where is `sortedWindows` defined? I think you changed that in the last commit and forgot to change here